.github/workflows/tfplugindocs-check.yml

```diff
@@ -0,0 +1,49 @@
+name: Check if tfplugindocs result matches /docs
+
+on:
+  push:
+    branches:
+      - master
+      - "release/**"
+  pull_request:
+    paths:
+      - .github/workflows/tfplugindocs-check.yml
+      - go.sum
+      - Makefile
+      - sonarqube/**
+      - main.go
+      - examples/**
+      - docs/**
+      - templates/**
+
+jobs:
+  tfplugindocs_check:
+    name: tfplugindocs check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # v4.2.1
```

Comment:
Is there a reason we want to pin these actions to the git sha instead of just the semantic version? The comments is helpful but i would prefer just the version straight up if that works.

Reply:
@jdamata I’d actually prefer to see this in all cases - I have seen two incidents recently where pinning to a major version caused issues.
- In one case, the action author did not respect semantic versioning and introduced a breaking change, which broke us
- In another case the action author accidentally deleted the version tag so the action “disappeared”

Using the commit SHA would have meant we were unaffected by these problems.

Using the SHA is also a recommended security hardening practice by github themselves: https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions#using-third-party-actions

Obviously your repo, your choice - but for me the above would be three good reasons to use the SHA

Reply:
I see. Good points, i can get behind using the git sha